src/interfacesAndTypes.tsx

```diff
@@ -4,42 +4,37 @@ export interface IStudentGroup {
   selectedStudent: ISelectedStudent
 }
 
-export interface IStudent {
-  studentName: string
-  studentId: string
+export interface ISelectedStudent {
   docId: string
-  profilePic?: string
-  studentFacts: IStudentFacts
+  studentId: string
+  studentInfo: ISelectedStudentInfo
+  studentGroupId: string
+  studentGroupName: string
+  selected: boolean
+  order: number
 }
 
-export interface INewStudentValues {
+interface ISelectedStudentInfo {
   studentName: string
   profilePic: string
-  [key: string]: string
-}
-
-export interface IStudentToAdd {
-  studentId: string
-  studentName: string
-  profilePic?: string
-  studentFacts: IStudentFacts
-}
-
-export interface IStudentFacts {
-  [key: string]: IStudentFact
+  selectedFact: IStudentFact | null
 }
 
 export interface IStudentFact {
   value: string
   title: string
 }
 
-export interface GroupParams {
-  groupId: string
+export interface IStudent {
+  studentName: string
+  studentId: string
+  docId: string
+  profilePic?: string
+  studentFacts: IStudentFacts
 }
 
-export interface StudentParams {
-  studentId: string
+export interface IStudentFacts {
+  [key: string]: string
 }
```

Comment:
It's always just one fact?

Reply:
No, not usually at least.  It's length can vary.  See the response below